server.js

```diff
@@ -6287,6 +6287,92 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// MicroBadger integration.
+camp.route(/^\/microbadger\/(image-size|layers)\/([^/]+)\/([^/]+)\/?([^/]*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var type = match[1];
+  var user = match[2];
+  var repo = match[3];
+  var tag = match[4];
+  var format = match[5];
+  if (user === '_') {
+    user = 'library';
+  }
+  var path = user + '/' + repo;
+  var url = 'https://api.microbadger.com/v1/images/' + path;
+
+  var badgeData = getBadgeData(type, data);
+  if (type === 'image-size') {
+    badgeData.text[0] = getLabel('image size', data);
+  }
+
+  var options = {
+    method: 'GET',
+    uri: url,
+    headers: {
+      'Accept': 'application/json'
+    }
+  };
+  request(options, function(err, res, buffer) {
+    if (res && res.statusCode === 404) {
+      badgeData.text[1] = 'not found';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    if (err != null || !res || res.statusCode !== 200) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    try {
+      var parsedData = JSON.parse(buffer);
+      var image;
+
+      if (tag) {
+        if (!parsedData['Versions']) {
```

Comment:
Does this project have a convention for dot-notation vs square-bracket on object properties, or we not care?

Reply:
I see we waiting for the answer of another maintainer but personally, I use square-bracket only when the key has an uppercase first letter.

Reply:
For historical reasons `server.js` has really lax rules, and [we're about to rewrite the services](https://github.com/badges/shields/projects/2) which will slowly empty it out. Elsewhere our code uses dots on any possible key (c.f. [eslint-config-standard](https://standardjs.com/)).